<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>EditClassDialog</class>
 <widget class="QDialog" name="EditClassDialog">
  <property name="windowModality">
   <enum>Qt::WindowModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>519</width>
    <height>446</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowIcon">
   <iconset resource="../resources/resources.qrc">
    <normaloff>:/topcased_icons/Class_24.gif</normaloff>:/topcased_icons/Class_24.gif</iconset>
  </property>
  <layout class="QFormLayout" name="formLayout">
   <item row="0" column="0">
    <widget class="QLabel" name="label">
     <property name="text">
      <string>Datatype name</string>
     </property>
     <property name="buddy">
      <cstring>nameLineEdit</cstring>
     </property>
    </widget>
   </item>
   <item row="0" column="1">
    <widget class="QLineEdit" name="nameLineEdit"/>
   </item>
   <item row="4" column="0" colspan="2">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
   <item row="3" column="0" colspan="2">
    <widget class="QGroupBox" name="groupBox">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
       <horstretch>0</horstretch>
       <verstretch>1</verstretch>
      </sizepolicy>
     </property>
     <property name="title">
      <string>Member variables</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="1">
       <widget class="QLineEdit" name="varNameLineEdit"/>
      </item>
      <item row="0" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Name</string>
        </property>
        <property name="buddy">
         <cstring>varNameLineEdit</cstring>
        </property>
       </widget>
      </item>
      <item row="4" column="0" colspan="2">
       <widget class="Gui::EditListWidget" name="editListWidget" native="true"/>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Type</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignTop</set>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="Gui::EditDatatypeNameWidget" name="editDatatypeNameWidget" native="true"/>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>Gui::EditListWidget</class>
   <extends>QWidget</extends>
   <header>Gui/EditListWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>Gui::EditDatatypeNameWidget</class>
   <extends>QWidget</extends>
   <header>Gui/EditDatatypeNameWidget.h</header>
   <container>1</container>
   <slots>
    <signal>datatypeEdited()</signal>
   </slots>
  </customwidget>
 </customwidgets>
 <tabstops>
  <tabstop>nameLineEdit</tabstop>
  <tabstop>varNameLineEdit</tabstop>
  <tabstop>buttonBox</tabstop>
 </tabstops>
 <resources>
  <include location="../resources/resources.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>EditClassDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>254</x>
     <y>439</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>EditClassDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>322</x>
     <y>439</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>varNameLineEdit</sender>
   <signal>textEdited(QString)</signal>
   <receiver>EditClassDialog</receiver>
   <slot>slotEdited()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>174</x>
     <y>139</y>
    </hint>
    <hint type="destinationlabel">
     <x>398</x>
     <y>156</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>editDatatypeNameWidget</sender>
   <signal>datatypeEdited()</signal>
   <receiver>EditClassDialog</receiver>
   <slot>slotEdited()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>302</x>
     <y>218</y>
    </hint>
    <hint type="destinationlabel">
     <x>259</x>
     <y>222</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>slotAdd()</slot>
  <slot>slotRemove()</slot>
  <slot>slotMoveUp()</slot>
  <slot>slotMoveDown()</slot>
  <slot>slotSelectedRowChanged()</slot>
  <slot>slotEdited()</slot>
 </slots>
</ui>
